public/store/AlertsStore.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { NotificationsStart } from 'opensearch-dashboards/public';
+import { AlertItem } from '../pages/Overview/models/interfaces';
+import { AlertsService, DetectorsService } from '../services';
+import { errorNotificationToast } from '../utils/helpers';
+import { GetAlertsResponse, ServerResponse } from '../../types';
+
+export class AlertsStore {
+  constructor(
+    private readonly service: AlertsService,
+    private readonly detectorsService: DetectorsService,
+    private readonly notifications: NotificationsStart
+  ) {}
+
+  public async getAlertsByDetector(detectorId: string, detectorName: string) {
```

Comment:
In Security analytics, alert object doesn't have the monitor name, so we need this. Also, detector name could change in the future since it is editable so it is better to not use it from the alert object.